sdk/Trace/ExporterFactory.php

```diff
@@ -0,0 +1,178 @@
+
+    public function __construct(string $name)
+    {
+        $this->name = $name;
+    }
+
+    /**
+      * Returns the coresponding Exporter via the configuration string
+      *
+      * @param string $configurationString String containing unextracted information for Exporter creation
+      * Should follow the format: contribType+baseUrl?option1=a
+      * Query string is optional and based on the Exporter
+      */
+    public function fromConnectionString(string $configurationString)
+    {
+        $strArr = explode('+', $configurationString);
+        // checks if input is given with the format type+baseUrl
+        if (sizeof($strArr) !== 2) {
+            throw new Exception('Invalid format.');
+        }
+
+        $contribName = strtolower($strArr[0]);
+        $endpointUrl = $strArr[1];
+
+        if (!$this->isAllowed($contribName)) {
+            throw new Exception('Invalid contrib name.');
+        }
+
+        // @phan-suppress-next-line PhanUndeclaredClassMethod
+        $dsn = empty($endpointUrl) ? '' : DsnParser::parse($endpointUrl);
+        $endpointUrl = $this->parseBaseUrl($dsn);
+        // parameters are only retrieved if there was an endpoint given
+        $args = empty($dsn) ? [] : $dsn->getParameters();
+        $scheme = empty($dsn) ? '' : $dsn->getScheme();
+        
+        switch ($contribName) {
+            case 'jaeger':
+                return $exporter = $this->generateJaeger($endpointUrl);
+            case 'zipkin':
+                return $exporter = $this->generateZipkin($endpointUrl);
+            case 'newrelic':
+                return $exporter = $this->generateNewrelic($endpointUrl, $args['licenseKey'] ?? null);
+            case 'otlp':
+                return $exporter = $this->generateOtlp($scheme);
+            case 'otlpgrpc':
+                return $exporter = $this->generateOtlpGrpc();
+            case 'zipkintonewrelic':
+                return $exporter = $this->generateZipkinToNewrelic($endpointUrl, $args['licenseKey'] ?? null);
+            default:
+                throw new Exception('Invalid contrib name.');
+            }
+    }
+
+    private function isAllowed(string $exporter)
+    {
+        return array_key_exists($exporter, $this->allowedExporters) && $this->allowedExporters[$exporter];
+    }
+
+    // constructs the baseUrl with the arguments retrieved from the raw baseUrl
+    private function parseBaseUrl($dsn)
+    {
+        if ($dsn == false) {
+            throw new Exception('Invalid endpoint');
+        }
+        $parsedUrl = '';
+        $parsedUrl .= empty($dsn->getScheme()) ? '' : $dsn->getScheme() . '://';
+        $parsedUrl .= empty($dsn->getHost()) ? '' : $dsn->getHost();
+        $parsedUrl .= empty($dsn->getPort()) ? '' : ':' . $dsn->getPort();
+        $parsedUrl .= empty($dsn->getPath()) ? '' : $dsn->getPath();
+
+        return $parsedUrl;
+    }
+
+    private function generateJaeger(string $endpointUrl)
```

Comment:
These look like they could be made part of the exporter?

Maybe we add a `Exporter::fromConnectionString()` to the interface, each exporter can then create itself (factory, named constructor), making this factory basically a lookup which exporter factory to use.

Reply:
Yeah this makes sense - I will update to reflect that

Reply:
@dkarlovi Should this change be part of this PR ? If i understood the change correctly, this would change the test cases and also the structure proposed in issue #184

Reply:
@alexperez52 IMO yes, it's a design choice to make this requirement apparent for all new exporters. It's up to you and @bobstrecansky